backend-lib/Cargo.lock

```diff
@@ -775,7 +760,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "969c513e03167e65d4bb59f5c51ec3820210975044ad7f218ab801fc169760fa"
 dependencies = [
  "base58",
- "hdwallet",
+ "hdwallet 0.3.1",
```

Comment:
`hdwallet` and `hdwallet-bitcoin` are in the same repo, so I'm confused that they aren't being bumped together. What depends on the older version of `hdwallet-bitcoin`?

Reply:
@str4d @nuttycom could you please assist us here?

Reply:
I can see that the iOS FFI uses version `0.4.1`. See [link](https://github.com/zcash-hackworks/zcash-light-client-ffi/blob/055b239fdd4218cdc9d7cbf354689f50891dc726/rust/Cargo.lock#L704). @str4d @nuttycom could you clarify please?

Reply:
@daira upgrading to `hdwallet 0.4.1` introduces a duplicate `secp256k1` dependency, and that's one that we can't upgrade (at least right now) - which do you prefer?

Reply:
I checked this, and upgrading is actually what we do need to do in this case, because it de-duplicates.

Reply:
This seems fine to me:
```diff
zcash-android-wallet-sdk/backend-lib$ git diff
diff --git a/backend-lib/Cargo.toml b/backend-lib/Cargo.toml
index c87cff94..765a57ae 100644
--- a/backend-lib/Cargo.toml
+++ b/backend-lib/Cargo.toml
@@ -12,8 +12,8 @@ rust-version = "1.60"
 
 [dependencies]
 failure = "0.1"
-hdwallet = "0.3.1"
-hdwallet-bitcoin = "0.3"
+hdwallet = "0.4.1"
+hdwallet-bitcoin = "0.4"
 hex = "0.4"
 jni = { version = "0.20", default-features = false }
 prost = "0.12"

zcash-android-wallet-sdk/backend-lib$ cargo update -p hdwallet-bitcoin
    Updating crates.io index
    Updating base58 v0.1.0 -> v0.2.0
    Removing block-buffer v0.7.3
    Removing block-padding v0.1.5
    Removing byte-tools v0.3.1
    Removing digest v0.8.1
    Removing generic-array v0.12.4
    Removing hdwallet v0.3.1
    Updating hdwallet-bitcoin v0.3.0 -> v0.4.1
    Removing opaque-debug v0.2.3
    Removing ripemd160 v0.8.0

zcash-android-wallet-sdk/backend-lib$ cargo update -p hdwallet
    Updating crates.io index
```

Reply:
Fixed. What I was remembering is that we don't want to upgrade to `secp256k1 0.27`, which is still the case.